File: ssluzba konacno/src/test/javascript/e2e/pages/nastavnici/nastavnici-creation.page.js
// Prazan konstruktor
// Posto je "element" globalna funkcija, ne moramo da prosledimo nista.
var NastavniciCreationPage = function() {};
var utils = require('../utils.js');

// Opis polja i metoda
NastavniciCreationPage.prototype = Object.create({}, {
    
    modalDialog: {
        get: function() {
            return utils.waitForElementPresence(by.className('modal-dialog'), 10000);
        }
    },

    modalTitle: {
        get: function() {
            return utils.waitForElementPresence(by.id('myNastavniciLabel'), 10000);
        }
    },

    ime: {
        get: function() {
            return utils.waitForElementPresence(by.name('ime'), 10000);
        },
        set: function(value) {
            this.ime.clear();
            this.ime.sendKeys(value);
        }
    },

    prezime: {
        get: function() {
            return utils.waitForElementPresence(by.name('prezime'), 10000);
        },
        set: function(value) {
            this.prezime.clear();
            this.prezime.sendKeys(value);
        }
    },

    zvanje: {
        get: function() {
            return utils.waitForElementPresence(by.name('zvanje'), 10000);
        },
        set: function(value) {
            this.zvanje.clear();
            this.zvanje.sendKeys(value);
        }
    },
    
    cancelButton: {
        get: function() {
            return this.modalDialog.element(by.className('btn-default'));
        }
    },
    
    saveBtn: {
        get: function() {
            return this.modalDialog.element(by.className('btn-primary'));
        }
    },

    createNastavnik: {
        value: function(ime, prezime, zvanje) {
            this.ime = ime;
            this.prezime = prezime;
            this.zvanje = zvanje;
            
            this.saveBtn.click();
        }
    }
});

// Export klase
module.exports = NastavniciCreationPage;

File: ssluzba konacno/src/main/java/rs/ac/uns/testdevelopment/ssluzba/pages/nastavnici/NastavniciCreationPage.java
package rs.ac.uns.testdevelopment.ssluzba.pages.nastavnici;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import helpers.Utils;

public class NastavniciCreationPage {
	private WebDriver driver;

	public NastavniciCreationPage(WebDriver driver) {
		super();
		this.driver = driver;
	}

	public WebElement getModalDialog() {
		return Utils.waitForElementPresence(driver, By.className("modal-dialog"), 10);
	}

	public WebElement getModalTitle() {
		return Utils.waitForElementPresence(driver, By.id("myNastavniciLabel"), 10);
	}

	public WebElement getIme() {
		return Utils.waitForElementPresence(driver, By.name("ime"), 10);
	}

	public void setIme(String value) {
		WebElement el = this.getIme();
		el.clear();
		el.sendKeys(value);
	}

	public WebElement getPrezime() {
		return Utils.waitForElementPresence(driver, By.name("prezime"), 10);
	}

	public void setPrezime(String value) {
		WebElement el = this.getPrezime();
		el.clear();
		el.sendKeys(value);
	}

	public WebElement getZvanje() {
		return Utils.waitForElementPresence(driver, By.name("zvanje"), 10);
	}

	public void setZvanje(String value) {
		WebElement el = this.getZvanje();
		el.clear();
		el.sendKeys(value);
	}

	public WebElement getCancelBtn() {
		return getModalDialog().findElement(By.className("btn-default"));
	}

	public WebElement getSaveBtn() {
		return getModalDialog().findElement(By.className("btn-primary"));
	}

	public void createNastavnik(String ime, String prezime, String zvanje) {
		setIme(ime);
		setPrezime(prezime);
		setZvanje(zvanje);
		getSaveBtn().click();
	}

}
